packages/patterns/src/components/reset-overrides-control.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	store as blockEditorStore,
+	BlockControls,
+} from '@wordpress/block-editor';
+import { ToolbarButton, ToolbarGroup } from '@wordpress/components';
+import { useSelect, useRegistry } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { parse } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+function recursivelyFindBlockWithId( blocks, id ) {
+	return blocks.find( ( block ) => {
+		if ( block.attributes.metadata?.id === id ) {
+			return block;
+		}
+
+		return recursivelyFindBlockWithId( block.innerBlocks, id );
+	} );
```

Comment:
@kevin940726 I don't think this does what's expected. `blocks.find` will only ever return a top-level block since it gets the value from the current array that the callback returns `truthy` for.

If you try a pattern where an overriden paragraph is nested in a group it doesn't work as expected.

Reply:
Oops! I don't know what I was thinking 😅. Fixed in [6ec77d7](https://github.com/WordPress/gutenberg/pull/57907/commits/6ec77d73346f147709e6ab434f8bb4335e4fcdb2). Hopefully I'm not missing something obvious again! 🤞